docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+
+## Pros and Cons of the Templating Options
+
+`fxa-auth-server` currently makes use of Mustache to develop email templates. Mustache is a logicless HTML templating framework and has been proven to be super helpful in developing mail components so far, but due to its logicless nature it becomes tedious to include conditional statements in the mail components. The following section will detail upon the various pros and cons of potential libraries/ frameworks that could be used to replace Mustache as a templating solution.
+
+### Option A - Continue to use Mustache
+
+- Description
+  - Keep the email templates as is in Mustache
+- Pros
+  - The effort to refactor all the templates would be eliminated for now
+- Cons
+  - The logicless nature of Mustache would continue to bother us in the future
+  - If we continue to use Mustache for our templates then our front-end frameworks in FxA would continue to be different. We currently have Backbone in the content-server, Mustache in the auth-server, and React everywhere else.
+
+### Option B - Use React server-side to generate static HTML templates
+
+- Description
+  - Template the auth server emails using [React](https://reactjs.org/) server-side and convert them over to [static HTML templates](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup). Other pros and cons of React can be found [in this ADR](https://github.com/mozilla/fxa/blob/main/docs/adr/0010-transition-fxa-from-backbone-to-react.md#option-c---move-to-isomorphic-react), though the type of React we're describing here would be stateless.
+- Pros
+  - The logicless nature of the templates will be eliminated once they're refactored in React
+  - We already use React in `fxa-settings`, `fxa-admin-panel`, `fxa-payments-server`, and `fxa-react`. There’s potential to cross-share components with the new email templates.
+  - React offers us many integration options. E.g there is one with [MJML](https://github.com/wix-incubator/mjml-react)
+  - React has already been audited by our security team.
+- Cons
+  - It could be a overkill to use React, since the library comes with far too many features which may not be needed for templating
+  - The templates in the `fxa-auth-server` do not contain heavy business logic so building them with a frontend-heavy framework like React might complicate things
+  - Since we will be converting React components into static HTML pages, another layer of complexity as well as new dependencies would be added to the project
```

Comment:
Does this need something like Next.js to server-side render them? What does the tooling options look like here and how will that be integrated into the auth-server?

Reply:
We shouldn't need to use NextJS since we don't need to keep client/server state in tact, this is more like we need to replace the Handlebars renderer with some build steps/processing, maybe with webpack or a task runner.